mlrun/projects/project.py

```diff
@@ -1011,6 +1017,7 @@ def set_artifact(
         self,
         key,
         artifact: typing.Union[str, dict, Artifact] = None,
+        artifact_kind: str = None,
         target_path: str = None,
         tag: str = None,
```

Comment:
this breaks compatibility, because this is user facing we can't add new params in the middle of the params as users that were dependent on the order will fail. add it as the last param